include/deal.II/lac/petsc_vector_base.h

```diff
@@ -848,8 +863,10 @@ namespace PETScWrappers
 
       const PetscInt petsc_i = index;
 
+      PetscScalar cast_value = static_cast<PetscScalar> (value);
+
       const int ierr
-        = VecSetValues (vector, 1, &petsc_i, &value, INSERT_VALUES);
+        = VecSetValues (vector, 1, &petsc_i, &cast_value, INSERT_VALUES);
```

Comment:
I don't think I understand this. `value` has type `const PetscScalar &`. `cast_value` has type `PetscScalar`. If you take the address, you would get `const PetscScalar *` and `PetscScalar *`, respectively. So the only difference is in the `const`. But `VecSetValues` takes a const pointer as fourth argument, as per http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Vec/VecSetValues.html . So it seems to me that you don't really need this change. Any insight?